### ISSUE TITLE
Feature request : live latex rendering

### ISSUE THREAD

**Issue by mocquin** [jupyterlab/jupyterlab#5884]
Hello there,

I recently noticed that https://math.stackexchange.com/ provides a live rendering of post's text, including latex equations : 

![latex](https://user-images.githubusercontent.com/36188368/51373556-c3c61780-1b00-11e9-977e-f763dc1c55f2.jpg)

This allows one to live-check latex formulas, which can be very handy when dealing with long-complicated/imbricated formulas.
I'm not sure how this kind of feature could implemented in jlab, but I think it could be very usefull.

**Comment by ian-r-rose** [comment]
I think this is a nice idea for an extension. The author could create a panel with an editor widget (possibly with LaTeX mode already loaded), and a rendered formula widget. Overall, I think it could be accomplished in ~200 lines of code.

**Comment by bollwyvl** [comment]
I am assuming this in in the Notebook, as markdown files already just work
this way, right? Synced scrolling and matching highlight would help for
sure, but you should get live rendering

Going a bit further, I started this one:

https://github.com/deathbeds/jupyterlab-outsource

Which concentrates on rich _editing_ with prosemirror. Definitely a work in
progress. Equation editing is possible, but not started yet, need to find
the right (supported) library and integrated with prosemirror.

Anyhow, it's like "create new view for output," but more "create new editor
for input". The core of it could handle live rendering trivially, "create
new view for input" as all the hard stuff is managing the sync problem.

Another way to think about the problem would be to tie the content of a
markdown cell to an ephemeral markdown file, and then render that. This is
kind of like the data bus stuff, but more general.

**Comment by mocquin** [comment]
Indeed, I meant for markdown cells in notebooks - updating my post ;)

**Comment by tangbinh** [comment]
I used JupyterLab to write lots of notebooks involving Math equations, so this feature would be a game-changer for me. Do we have any idea if / when it will be supported?

**Comment by jasongrout** [comment]
Here's one interesting relevant extension that is under development: https://github.com/jupytercalpoly/jupyterlab-richtext-mode

**Comment by tangbinh** [comment]
> Here's one interesting relevant extension that is under development: https://github.com/jupytercalpoly/jupyterlab-richtext-mode

Thank you for the link. The extension is surely interesting. However, I found it very unstable ; equations seem to jump around in a mysterious way, some lost and some not rendered properly. It would be great if we can have something similar to [Typora](https://www.typora.io) or [Mark Text](https://github.com/marktext/marktext), both of which follow the WYSIWYG philosophy. Even just a Markdown preview mode for individual cells would be good. In this regard, I think [Google Colab](https://colab.research.google.com/notebooks/markdown_guide.ipynb) does a very good job, and I think a lot of people would appreciate this functionality if it is included in JupyterLab as well.

**Comment by knifelees3** [comment]
I really think this feature should be added to jupyter lab. jupyter notebook has support this and is really helpful when we need type some long equations.

**Comment by jasongrout** [comment]
> jupyter notebook has support this and is really helpful when we need type some long equations.

The stock Jupyter Notebook does not live render markdown, right? Are you talking about a notebook extension that enables live preview?

**Comment by knifelees3** [comment]
Yes, Jupyter Notebook need extensions to support live preview and Jupyter
Lab doesn't have this kind of extensions

Jason Grout <notifications@github.com> 于2020年3月30日周一 下午11:22写道：

**Comment by jasongrout** [comment]
Thanks. I'll label this issue as an extension idea, then, to provide a live preview.

**Comment by jasongrout** [comment]
> Thanks. I'll label this issue as an extension idea, then, to provide a live preview.

Already done, actually, last year by Ian. If someone sees this and starts working on it, we'd love if you'd post here to let people know!

**Comment by dhirschfeld** [comment]
nteract live previews your markdown below the cell as you type it - there might be something to crib from there?

**Comment by ichsan2895** [comment]
> Hello there,
> 
> I recently noticed that https://math.stackexchange.com/ provides a live rendering of post's text, including latex equations :
> 
> ![latex](https://user-images.githubusercontent.com/36188368/51373556-c3c61780-1b00-11e9-977e-f763dc1c55f2.jpg)
> 
> This allows one to live-check latex formulas, which can be very handy when dealing with long-complicated/imbricated formulas.
> I'm not sure how this kind of feature could implemented in jlab, but I think it could be very usefull to be able to live-preview markdown cells in notebooks (kinda like markdown files).

Hello, Try to use Jupyter LaTex.
1. click "new launcer"
2. select "Markdown file"
3. Type something like $x_1$
4. Right click and select "Show markdown preview".

Unfortunately it doesn't available in notebook yet, so just copy the text (from "markdown" tab) to notebook

**Comment by erentar** [comment]
Currently it is extremely undergonomic to type equations in markdown cells without seeing a preview. This feature is urgently needed.

**Comment by jasongrout** [comment]
We welcome people to work on this! If someone can work on this, please post here. We're happy to answer questions to help them get started.

**Comment by maddox05** [comment]
since 2019 is crazy
